packages/data/src/test/registry.js

```diff
@@ -396,7 +396,7 @@ describe( 'createRegistry', () => {
 			return promise;
 		} );
 
-		it( 'should resolve promise non-action to dispatch', ( done ) => {
+		it( 'should resolve promise non-action to dispatch', async () => {
```

Comment:
Technically, since there's no `await`, we don't need `async` here, even if we are returning a promise.

```suggestion
		it( 'should resolve promise non-action to dispatch', () => {
```

We could await the promise instead of returning it. I seem to recall that the Jest team encourages that a promise be returned when possible. Not entirely sure the rationale, but it sticks in my mind 😄